### ISSUE TITLE
Tests are broken with pyparsing 2.4.0

### ISSUE THREAD

**Issue by sgillies** [mapbox/snuggs#15]
For now, I'm pinning pyparsing at 2.3.1 in the Travis builds, but we'll need to get the tests to pass with 2.4 sooner or later.

**Comment by QuLogic** [comment]
I think these are the test failures you mean?
```pytb
__________________________ test_missing_closing_paren __________________________
    def test_missing_closing_paren():
        with pytest.raises(SyntaxError) as excinfo:
            snuggs.eval("(+ 1 2")
        assert excinfo.value.lineno == 1
        assert excinfo.value.offset == 7
        exception_options = ['expected a function or operator',
                             'Expected {Forward: ... | Forward: ...}']
>       assert str(excinfo.value) in exception_options
E       assert 'Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - | & | | | <= | <...quoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}' in ['expected a function or operator', 'Expected {Forward: ... | Forward: ...}']
E        +  where 'Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - | & | | | <= | <...quoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}' = str(ExpressionError('Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - ...uoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}'))
E        +    where ExpressionError('Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - ...uoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}') = <ExceptionInfo ExpressionError tblen=3>.value
test_snuggs.py:184: AssertionError

______________________________ test_missing_func2 ______________________________
    def test_missing_func2():
        with pytest.raises(SyntaxError) as excinfo:
            snuggs.eval("(# 1 2)")
        assert excinfo.value.lineno == 1
        assert excinfo.value.offset == 2
        exception_options = ['expected a function or operator',
                             'Expected {Forward: ... | Forward: ...}']
>       assert str(excinfo.value) in exception_options
E       assert 'Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - | & | | | <= | <...quoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}' in ['expected a function or operator', 'Expected {Forward: ... | Forward: ...}']
E        +  where 'Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - | & | | | <= | <...quoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}' = str(ExpressionError('Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - ...uoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}'))
E        +    where ExpressionError('Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - ...uoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}') = <ExceptionInfo ExpressionError tblen=3>.value
test_snuggs.py:202: AssertionError

_________________________ test_bogus_higher_order_func _________________________
    def test_bogus_higher_order_func():
        with pytest.raises(SyntaxError) as excinfo:
            snuggs.eval("((bogus * 2) 2)")
        assert excinfo.value.lineno == 1
        assert excinfo.value.offset == 3
        exception_options = ['expected a function or operator',
                             'Expected {Forward: ... | Forward: ...}']
>       assert str(excinfo.value) in exception_options
E       assert 'Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - | & | | | <= | <...quoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}' in ['expected a function or operator', 'Expected {Forward: ... | Forward: ...}']
E        +  where 'Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - | & | | | <= | <...quoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}' = str(ExpressionError('Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - ...uoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}'))
E        +    where ExpressionError('Expected {Forward: Group:({{{{Suppress:("(") map | partial} {{{"nil" | Forward: ...} | * | + | / | - ...uoted string, starting with \' ending with \' | quoted string, starting with " ending with "}}]...} Suppress:(")")})}') = <ExceptionInfo ExpressionError tblen=3>.value
test_snuggs.py:220: AssertionError
```
We have this version on Rawhide already, so need to figure out what to do about it.

**Comment by QuLogic** [comment]
Oh, I was wrong, it's not just Rawhide, but F30 as well.

**Comment by QuLogic** [comment]
Any thoughts here?

**Comment by QuLogic** [comment]
Another little ping, please?

**Comment by sgillies** [comment]
@QuLogic sorry, too busy to get to this right now.

**Comment by tigerfoot** [comment]
snugs will be removed from openSUSE Tumbleweed (then from Leap release) due to its constant failure to make test. That's too bad.

**Comment by sgillies** [comment]
@tigerfoot bad enough for you that you would help? What are you trying to achieve with your comment?

**Comment by sgillies** [comment]
Resolved by 9e4827e.
